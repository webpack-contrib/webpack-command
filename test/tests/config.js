const { test } = require('../util');
const { distill } = require('../../lib/config');

const config = [
  {
    entry: resolve(__dirname, '../../common/entry-a.js'),
    mode: 'development',
  },
  {
    entry: resolve(__dirname, '../../common/entry-b.js'),
    mode: 'development',
  },
];

test('lib/config', module, () => {
  it(`distill()`, () => {
    const result = distill({}, config, {});
    expect(result).toMatchSnapshot();
  });

  it(`distill() plugins`, () => {
    const result = distill({}, { plugins: [] }, { plugins: [1] });
    expect(result).toMatchSnapshot();
  });

  it(`distill() plugins from config array`, () => {
    const conf = [].concat(config);
    conf[0].plugins = [];
    const result = distill({}, conf, { plugins: [1] });
    expect(result).toMatchSnapshot();
  });

  it(`distill() --config-name`, () => {
    config[1].name = 'test';

    const argv = { configName: 'test' };
    const result = distill(argv, config, {});
    expect(result).toMatchSnapshot();

    delete config[1].name;
  });

  it(`distill() --config-name not found`, () => {
    const argv = { configName: 'test' };
    const stub = () => distill(argv, config, {});
    expect(stub).toThrow();
  });

  it(`distill() --config-name not array`, () => {
    const argv = { configName: 'test' };
    const stub = () => distill(argv, config[0], {});
    expect(stub).toThrow();
  });
});
