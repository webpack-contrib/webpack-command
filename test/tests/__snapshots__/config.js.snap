// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/config > distill() #0 1`] = `
Array [
  Object {
    "entry": "<PROJECT_ROOT>/common/entry-a.js",
    "mode": "development",
  },
  Object {
    "entry": "<PROJECT_ROOT>/common/entry-b.js",
    "mode": "development",
  },
]
`;

exports[`lib/config > distill() --config-name #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/common/entry-b.js",
  "mode": "development",
  "name": "test",
  "plugins": Array [
    1,
  ],
}
`;

exports[`lib/config > distill() plugins #0 1`] = `
Object {
  "plugins": Array [
    1,
  ],
}
`;

exports[`lib/config > distill() plugins #1 1`] = `
Object {
  "plugins": Array [
    1,
  ],
}
`;

exports[`lib/config > distill() plugins #2 1`] = `
Object {
  "plugins": Array [
    1,
    1,
  ],
}
`;

exports[`lib/config > distill() plugins from config array #0 1`] = `
Array [
  Object {
    "entry": "<PROJECT_ROOT>/common/entry-a.js",
    "mode": "development",
    "plugins": Array [
      1,
    ],
  },
  Object {
    "entry": "<PROJECT_ROOT>/common/entry-b.js",
    "mode": "development",
    "plugins": Array [
      1,
    ],
  },
]
`;

exports[`lib/config > distill() plugins from config array #1 1`] = `
Array [
  Object {
    "entry": "<PROJECT_ROOT>/common/entry-a.js",
    "mode": "development",
    "plugins": Array [
      1,
    ],
  },
  Object {
    "entry": "<PROJECT_ROOT>/common/entry-b.js",
    "mode": "development",
    "plugins": Array [
      1,
    ],
  },
]
`;

exports[`lib/config > distill() plugins from config array #2 1`] = `
Array [
  Object {
    "entry": "<PROJECT_ROOT>/common/entry-a.js",
    "mode": "development",
    "plugins": Array [
      1,
    ],
  },
  Object {
    "entry": "<PROJECT_ROOT>/common/entry-b.js",
    "mode": "development",
    "plugins": Array [
      1,
      1,
    ],
  },
]
`;
